tests/integration/targets/rds_instance/roles/rds_instance/tasks/test_upgrade.yml

```diff
@@ -62,6 +62,46 @@
         password: "{{ password }}"
         db_instance_class: "{{ db_instance_class }}"
         allocated_storage: "{{ allocated_storage }}"
+        apply_immediately: True
+      register: result
+
+    - assert:
+        that:
+          - result.changed
+          - '"engine_version" in result.pending_modified_values or result.engine_version == mariadb_engine_version_2'
+
+    ### Specifying allow_major_version_upgrade with check_mode will always result in changed=True
```

Comment:
Your aren't specifying `allow_major_version_upgrade`.

Reply:
That was intentional. The comment there was saying "hey, I'm not testing allow_major_version_upgrade, because if I did, check mode idempotence test would fail"

Reply:
But I guess it more clear if you remove the assert and use  `allow_major_version_upgrade ` as a parameter and leave the comment explaining the issue or you can comment the expected assert - not changed.

Reply:
that makes sense. ill update thanks